### PR TITLE
Make test

### PR DESCRIPTION
**Pull request by majikandy**
# `Make test` to run tests with env variables

### Acceptance Criteria

- [x] _`make test` runs tests with .env and .env.overrides_
- [x] _`make test TEST=test_name` as above but filtered to run tests that start with the name_
- [ ] ~Bonus: `make source-env` if you just want to put env variables in your shell session to use for something else~ Note this turned out to be not possible as the child process can't set the shell variable in the parent

### Testing information

As above

### Pre-flight Checklist

- [ ] Devs can use it

### Pre-merge Checklist

- [ ] Branch test approved

[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"

### CODE REVIEW

File: medicines/doc-index-updater/Makefile [hunk 1]
```diff
@@ -6,6 +6,14 @@ tag := latest
 default: ## run locally
 	export $$(cat .env .env.overrides 2> /dev/null | xargs) && cargo run
 
+.PHONY: test
+test: ## run locally
+	export $$(cat .env .env.overrides 2> /dev/null | xargs) && cargo test $$TEST
```
Comment (craiga) [inline medicines/doc-index-updater/Makefile:11]: Can we document this behaviour somewhere? Add a "To run tests" section in the readme?

File: medicines/doc-index-updater/Makefile [hunk 2]
```diff
@@ -6,6 +6,14 @@ tag := latest
...
+
+.PHONY: source-env
+source-env: ## run locally
+	export $$(cat .env .env.overrides 2> /dev/null | xargs)
```
Comment (craiga) [inline medicines/doc-index-updater/Makefile:15]: :+1:

**Comment by majikandy** [comment]
I would suggest that the make file is part of the documentation isn’t it?

Alternatively 

We could make a ‘make help’ ?

File: medicines/doc-index-updater/Makefile (hunk 1 above)
Comment (majikandy) [reply medicines/doc-index-updater/Makefile:11]: I would suggest that the make file is part of the documentation isn’t it?

Alternatively

We could make a ‘make help’ ?

Comment (majikandy) [reply medicines/doc-index-updater/Makefile:11]: ah `make help` does this already - I wondered what #run locally was all about

Comment (majikandy) [reply medicines/doc-index-updater/Makefile:11]: @craiga do `make help` to see this in action with the latest of this branch

File: medicines/doc-index-updater/Makefile [hunk 3]
```diff
@@ -51,4 +55,4 @@ docker-push: ## Push Docker image
 
 .PHONY: help
 help: ## Display this help screen
-	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
\ No newline at end of file
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```
Comment (rjkerrison) [inline medicines/doc-index-updater/Makefile:58]: wow

Comment (rjkerrison) [reply medicines/doc-index-updater/Makefile:58]: this is great but it's not obvious it's here just from browsing the repo, so I agree a _To run tests_ section in the README would be great

Comment (majikandy) [reply medicines/doc-index-updater/Makefile:58]: Done

File: medicines/doc-index-updater/Makefile (hunk 1 above)
Comment (majikandy) [reply medicines/doc-index-updater/Makefile:11]: Did the readme entry too

**Pull request merged**